src/puzzle/core.cljs

```diff
@@ -25,32 +26,34 @@
 
 (defn- sprite-on-click
   [sprite]
-  (println (:name sprite)))
+  (println (:name sprite))
+  (println (get-in @game-state [:piece-coords (:name sprite)])))
 
 (defn ^:private init-board!
   "Create randomized puzzle board with one black piece"
   [game]
   (let [game-object-factory (:add game)
-        board-rows          (/ (:world-width @game-state) (:piece-width @game-state))
-        board-cols          (/ (:world-height @game-state) (:piece-height @game-state))
+        board-cols          (/ (:world-width @game-state) (:piece-width @game-state))
+        board-rows          (/ (:world-height @game-state) (:piece-height @game-state))
         shuffled-frame-nums (shuffle (range (* board-rows board-cols)))]
-    (doseq [col (range board-cols)
-            row (range board-rows)]
-      (let [frame-number (shuffled-frame-nums (+ (* col board-rows) row))
+    (doseq [row (range board-rows)
+            col (range board-cols)]
+      (let [frame-number (shuffled-frame-nums (+ (* row board-cols) col))
             piece        (if (= 1 frame-number)
                            (pgof/sprite game-object-factory
-                                        (* row (:piece-width @game-state))
-                                        (* col (:piece-height @game-state)))
+                                        (* col (:piece-width @game-state))
+                                        (* row (:piece-height @game-state)))
                            (pgof/sprite game-object-factory
-                                        (* row (:piece-width @game-state))
-                                        (* col (:piece-height @game-state))
+                                        (* col (:piece-width @game-state))
+                                        (* row (:piece-height @game-state))
                                         "logo"
                                         frame-number))]
-        (pset! piece :name (str "piece" frame-number))
+        (pset! piece :name (str frame-number))
+        (swap! game-state assoc-in [:piece-coords (:name piece)] [row col])
```

Comment:
(swap! game-state assoc-in [:piece-coords (:name piece)] [1 3])
=> {:piece-coords {"piece5" [1 3]}}